l10n_ch_payment_slip/invoice.py

```diff
@@ -67,15 +66,15 @@ def _is_generate_bvr(self, cr, uid, invoice, context=None):
         return (invoice.partner_bank_id and
                 invoice.partner_bank_id.state == 'bvr')
 
-    def _compute_bvr_ref(self, cr, uid, move_line, context=None):
+    def _compute_bvr_ref(self, cr, uid, move_line,
+                         sequence_int=0, context=None):
         ''' Default BVR reference computation '''
         res = ''
         if move_line.invoice.partner_bank_id.bvr_adherent_num:
             res = move_line.invoice.partner_bank_id.bvr_adherent_num
-        move_number = ''
-
+        move_number = str(sequence_int)
         if move_line.invoice.number:
-            compound = str(move_line.invoice.number) + str(move_line.id)
+            compound = str(move_line.invoice.number) + str(sequence_int)
```

Comment:
Yes, It fragile, but we need to have a number that must be identical if you cancel and reconfirmed the invoice.
as soon as the move line is different each time you reopen the invoice, you will get a different number, if you have already send the BVR report to your customer, you will not be able to match the payment with the invoice automatically